File: API/models/group_log.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, func, CheckConstraint
from models.base import Base


class GroupLog(Base):
    __tablename__ = "group_logs"

    id = Column(Integer, primary_key=True)
    group_id = Column(Integer, ForeignKey("groups.id", ondelete="CASCADE"), nullable=False)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    action = Column(String(20), nullable=False)  # JOIN, LEAVE
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    __table_args__ = (
        CheckConstraint("action IN ('JOIN', 'LEAVE')", name="chk_group_log_action"),
    )

File: Mobile/androidApp/src/main/java/com/example/budgeting/android/data/model/GroupLog.kt
package com.example.budgeting.android.data.model

import com.squareup.moshi.Json

data class GroupLog(
    @Json(name = "id")
    val id: Int,
    
    @Json(name = "group_id")
    val group_id: Int,
    
    @Json(name = "user_id")
    val user_id: Int,
    
    @Json(name = "action")
    val action: String, // "JOIN" or "LEAVE"
    
    @Json(name = "created_at")
    val created_at: String
)

